tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -296,26 +294,28 @@ private void FeaturesProductConfigChanged(object sender, ProductConfigChangedEve
             var features = e.GetDeserializedConfigurations<AsmFeatures>().FirstOrDefault();
             if (features.TypedFile != null)
             {
-                _settings.Enabled = features.TypedFile.Asm.Enabled;
-                UpdateStatus(true);
+                lock (_settings)
+                {
+                    _settings.Enabled = features.TypedFile.Asm.Enabled;
+                    UpdateStatus(true);
```

Comment:
I think this locks `_settings` twice in one callstack, is that safe?

Reply:
Yes, the same thread does not get locked by an owned lock